### PR TITLE
Matching torch.stft and torch.istft

### PR DESCRIPTION
**Pull request by mpariente**
Those are just first steps but it seems it won't be too hard. 

Until now: 
- STFT seems to pass easily, with various windows and window sizes. (More tests needed to cover edge cases etc..)
- ISTFT passes with square window (without counting the first and last frames where the overlap is not complete). 

More work needs to be done to support w**2 OLA correction after the ISTFT (highpri),  `center` and `normalize` (lowpri). 

`onesided` will always supposed to be `True` and we won't support the `onesided` argument. 

Would solve #1

### CODE REVIEW

**Comment by mpariente** [comment]
Cc @Baldwin-disso and @faroit

**Comment by mpariente** [comment]
The ISTFT looks in a good shape as well. I need to write more informal tests with different windows to make sure it works, but given the C++ code, I'm pretty sure it'll be fine.

**Comment by mpariente** [comment]
`torch.isttf` raises RuntimeError for most windows, so we need to use `center=True` to avoid boundary effects on the w**2 OLA. 
The forward works with `center=True` apparently (to be check with other strides etc.. but looks in a good shape)

**Comment by jonashaag** [comment]
FYI there is a ton of tests here: https://github.com/pytorch/pytorch/blob/master/test/test_spectral_ops.py probably overkill to duplicate all of them but maybe for inspiration! :)

**Comment by mpariente** [comment]
Thanks for that, it will be a good source of inspiration indeed.

**Comment by mpariente** [comment]
So the ISTFT also works with `center=True`, when we don't have the `length` argument.. And now comes a new problem, we should be able to pass kwargs to the hooks in order to pass the `length`. But how to organize that?

File: asteroid_filterbanks/enc_dec.py [hunk 1]
```diff
@@ -265,13 +274,14 @@ def forward(self, spec) -> torch.Tensor:
         """
         filters = self.get_filters()
         spec = self.filterbank.pre_synthesis(spec)
-        return multishape_conv_transpose1d(
+        wav = multishape_conv_transpose1d(
```
Comment (mpariente) [inline asteroid_filterbanks/enc_dec.py:277]: We need to pass a `kwargs` to `post_synthesis` in order to support the `length` argument. But it seems unfair to pass it only to `post_synthesis`.
Three ugly ways to do: 
- Pass all kwargs to each method `pre_synthesis`, `post_synthesis` (need `**kwargs` in the child hooks to accept unknown arguments).
- Pass `pre_kwargs` and `post_kwargs` dict to `forward` (super ugly on the user side)
- Receive all `**kwargs`, analyze the signature of both functions and pass accordingly. This looks like the most correct way to do. It's not too hard to implement but I wonder if it's worth it. 
Thoughts @jonashaag

Comment (jonashaag) [reply asteroid_filterbanks/enc_dec.py:277]: Shouldn't length be part of the decoder? It's useful for other filter banks as well, right? And I think the trimming/padding logic is the same for other filter banks (trim or zero pad [centered or not centered]).

Comment (mpariente) [reply asteroid_filterbanks/enc_dec.py:277]: I just thought about this possibility as well, my bad. 
So yes, this will be much better.

File: notebooks/torch_compat.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        view_shape = [1 for _ in wav.shape[:-1]] + [-1]
+        wsq_ola = wsq_ola.view(view_shape)
+
+        start = self.kernel_size // 2 if self.center else 0
+        wav = wav[..., start:]
+        wsq_ola = wsq_ola[..., start:]
+
+        min_mask = wsq_ola.abs() < 1e-11
+        if min_mask.any():
+            # Warning instead of error. Might be trimmed afterward.
```
Comment (popcornell) [inline notebooks/torch_compat.py:95]: the trimming is done only if length is used correct ?

Comment (mpariente) [reply notebooks/torch_compat.py:95]: In `torch.istft`

> The :attr:`n_fft`, :attr:`hop_length`, :attr:`win_length` are all the same which prevents the calculation
of right padding. These additional values could be zeros or a reflection of the signal so providing
:attr:`length` could be useful. If :attr:`length` is ``None`` then padding will be aggressively removed
(some loss of signal).

So, if there is no `length`, the output signal is always smaller than the input, which is weird for me.

Comment (faroit) [reply notebooks/torch_compat.py:95]: even of the signal has a length of `% win_lenght == 0` ?

Comment (mpariente) [reply notebooks/torch_compat.py:95]: No, indeed. In this case, the signal is the same.

**Comment by mpariente** [comment]
Finally passing!
Only works for `kernel_size == n_filters`. 
I won't spend more time on this.

**Comment by faroit** [comment]
@mpariente thanks will test that soon
